src/test/CharterManager-unit.t.sol

```diff
@@ -0,0 +1,644 @@
+        VatLike(manager.vat()).flux(adapter.ilk(), src, dst, wad);
+    }
+    function quit() public {
+        manager.quit(adapter.ilk(), address(this));
+    }
+    function quit(address dst) public {
+        manager.quit(adapter.ilk(), dst);
+    }
+    function hope(address vat, address usr) public {
+        Vat(vat).hope(usr);
+    }
+    function nope(address vat, address usr) public {
+        Vat(vat).nope(usr);
+    }
+
+    function try_call(address addr, bytes calldata data) external returns (bool) {
+        bytes memory _data = data;
+        assembly {
+            let ok := call(gas(), addr, 0, add(_data, 0x20), mload(_data), 0, 0)
+            let free := mload(0x40)
+            mstore(free, ok)
+            mstore(0x40, add(free, 32))
+            revert(free, 32)
+        }
+    }
+    function can_call(address addr, bytes memory data) internal returns (bool) {
+        (bool ok, bytes memory success) = address(this).call(
+            abi.encodeWithSignature(
+                "try_call(address,bytes)"
+            , addr
+            , data
+            ));
+        ok = abi.decode(success, (bool));
+        if (ok) return true;
+    }
+    function can_exit(address usr, uint256 val) public returns (bool) {
+        bytes memory call = abi.encodeWithSignature
+        ("exit(address,address,uint256)", address(adapter), usr, val);
+        return can_call(address(manager), call);
+    }
+}
+
+contract CharterManagerTest is TestBase {
+
+    Token               gem;
+    Token               dai;
+    Vat                 vat;
+    Vow                 vow;
+    Jug                 jug;
+    Spotter             spotter;
+    DSValue             pip;
+    DaiJoin             daiJoin;
+    ManagedGemJoin      adapter;
+    CharterManagerImp   manager;
+
+    address             self;
+    bytes32             ilk = "TOKEN-A";
+
+    uint256 constant NIB_ONE_PCT = 1.0 * 1e16;
+    uint256 constant CEILING     = 100 * 1e18;
+
+    function setUp() public virtual {
+        self = address(this);
+        gem = new Token(6, 1000 * 1e6);
+
+        // standard Vat setup
+        vat = new Vat();
+        vow = new Vow(address(vat), address(0), address(0));
+
+        jug = new Jug(address(vat));
+        jug.file("vow", address(vow));
+        vat.rely(address(jug));
+
+        spotter = new Spotter(address(vat));
+        vat.rely(address(spotter));
+
+        dai = new Token(18, CEILING);
+        daiJoin = new DaiJoin(address(vat), address(dai));
+        vat.rely(address(daiJoin));
+
+        vat.init(ilk);
+        vat.file("Line", 100 * CEILING * 1e27);
+        vat.file(ilk, "line", CEILING * 1e27);
+
+        jug.init(ilk);
+        uint256 TWO_PCT = 1000000000627937192491029810;
+        jug.file(ilk, "duty", TWO_PCT);
+
+        pip = new DSValue();
+        pip.poke(bytes32(uint256(1e27)));
+
+        spotter.file(ilk, "mat", RAY);
+        spotter.file(ilk, "pip", address(pip));
+        spotter.poke(ilk);
```

Comment:
This setup is also unnecessary--just `file` a `spot` value for the `ilk` directly into the `Vat`. Then `pip` and `spotter` can be eliminated (including the imports).

Reply:
Done.